server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -344,6 +345,18 @@ private void tryAllHandlers(final RestRequest request, final RestChannel channel
                         BytesRestResponse.
                             createSimpleErrorResponse(channel, BAD_REQUEST, "multiple values for single-valued header [" + name + "]."));
                     return;
+                } else if (name.equals(Task.TRACE_PARENT)) {
+                    String traceparent = distinctHeaderValues.get(0);
+                    // TODO rely on high performance `TraceContext` from apm-agent-java
+                    // This does little validation of the actual id values, since we only use them to persist in logging for now this
+                    // should be fine. Once we actually start instrumenting through apm-agent-java it will take care of validation.
+                    // Java has a fast path for single character regexes so this should not be as bad as its signature implies
+                    String[] tokens  = traceparent.split("-");
+                    if (tokens.length == 4) {
+                        threadContext.putTransient(Task.TRACE_ID, tokens[1]);
+                        threadContext.putTransient(Task.TRANSACTION_ID, tokens[2]);
```

Comment:
This is not the transaction ID, but rather the span id of the parent (the client call). We could either generate a new random ID, introduce a `parent.id` field (which we don't do yet for logs), or just leave it out.

Reply:
Aye, I left it as the transaction to not have information loss for now, although `span.id` would have been better for that. 



Relates: https://github.com/elastic/ecs-dotnet/pull/115 where we pushed back on `parent.id`.

Reply:
> Aye, I left it as the transaction to not have information loss for now

Got it. The issue is that it's the wrong information. This is not the `tranaction.id`.
I think it's fine to leave out the `transaction.id` because the log correlation feature works only off of the `trace.id`.